hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+        """
+        self._setup_tx_streaming()
+        start_from_hexlist = [tx.hex() for tx in start_from]
+        until_first_block_hex = until_first_block.hex()
+        self.log.debug('send_get_transactions_bfs', start_from=start_from_hexlist, last_block=until_first_block_hex)
+        payload = json.dumps(dict(
+            start_from=start_from_hexlist,
+            until_first_block=until_first_block_hex,
+        ))
+        self.send_message(ProtocolMessages.GET_TRANSACTIONS_BFS, payload)
+        self.receiving_stream = True
+
+    def handle_get_transactions_bfs(self, payload: str) -> None:
+        if self._is_streaming:
+            self.log.warn('ignore GET-TRANSACTIONS-BFS, already streaming')
+            return
+        data = json.loads(payload)
+        # XXX: todo verify this limit while parsing the payload.
+        start_from = data['start_from']
+        if len(start_from) > MAX_GET_TRANSACTIONS_BFS_LEN:
+            self.log.error('too many transactions in GET-TRANSACTIONS-BFS', state=self.state)
+            self.protocol.send_error_and_close_connection('Too many transactions in GET-TRANSACTIONS-BFS')
+            return
+        self.log.debug('handle_get_transactions_bfs', **data)
+        start_from = [bytes.fromhex(tx_hash_hex) for tx_hash_hex in start_from]
+        until_first_block = bytes.fromhex(data['until_first_block'])
+        self.send_transactions_bfs(start_from, until_first_block)
+
+    def send_transactions_bfs(self, start_from: list[bytes], until_first_block: bytes) -> None:
+        start_from_txs = []
+        for start_from_hash in start_from:
+            try:
+                start_from_txs.append(self.tx_storage.get_transaction(start_from_hash))
+            except TransactionDoesNotExist:
+                # In case the tx does not exist we send a NOT-FOUND message
+                self.log.debug('requested start_from_hash not found', start_from_hash=start_from_hash.hex())
+                self.send_message(ProtocolMessages.NOT_FOUND, start_from_hash.hex())
+                return
+        if not self.tx_storage.transaction_exists(until_first_block):
+            # In case the tx does not exist we send a NOT-FOUND message
+            self.log.debug('requested until_first_block not found', until_first_block=until_first_block.hex())
+            self.send_message(ProtocolMessages.NOT_FOUND, until_first_block.hex())
+            return
+        if self.transactions_streaming is not None and self.transactions_streaming.is_running:
+            self.transactions_streaming.stop()
+        self.transactions_streaming = TransactionsStreaming(self, start_from_txs, until_first_block,
+                                                            limit=self.DEFAULT_STREAMING_LIMIT)
+        self.transactions_streaming.start()
+
+    def send_transaction(self, tx: Transaction) -> None:
+        """Send a TRANSACTION message."""
+        # payload = bytes(tx).hex()  # fails for big transactions
```

Comment:
The p2p protocol is a line protocol and it has a maximum line length. We should confirm that the maximum vertex size fits any line otherwise sync would stall.

Reply:
The line protocol imposes a maximum line length to prevent DoS attacks. As discussed, this is not an issue.